src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,803 @@
+            ${cellWidth === 0 ? 'display: none' : ''}
+            ${columnSettings.allowOverflow ? 'overflow: visible;' : ''}
+            `}
+        >
+          {cellData}
+        </div>
+      );
+
+      cellIndex++;
+    });
+
+    row.push(
+      <div
+        id={rowId}
+        class={`
+        ${rowClass}
+        ${this.selectedRows.includes(rowId) ? ACTIVE_CLASS : ''}
+        `}
+        role="row"
+      >
+        {rowCells}
+      </div>
+    );
+
+    if (bodyRow.accordion) {
+      row.push(
+        <transition name="slide-fade">
+          <div
+            v-show={this.accordionsShown.includes(rowId)}
+            id={`row-${rowId}-content`}
+          >
+            {rowAccordionContent}
+          </div>
+        </transition>
+      );
+    }
+
+    return row;
+  }
+
+  body() {
+    let tableBodyRows: JSX.Element;
+
+    if (this.data.body.length > 0) {
+      const dataToRender = this.dataToRender();
+
+      tableBodyRows = dataToRender.map((bodyRow: DataTableRow) => {
+        return this.row(bodyRow);
+      });
+    } else {
+      const noResultsMessage = this.data.config.noResultsMessage
+        ? this.data.config.noResultsMessage
+        : 'No results found';
+
+      tableBodyRows = (
+        <div class={DATA_TABLE_CLASSES.EMPTY}>
+          <div>{noResultsMessage}</div>
+        </div>
+      );
+    }
+
+    return <div class={DATA_TABLE_CLASSES.BODY}>{tableBodyRows}</div>;
+  }
+
+  pagination() {
+    const results = this.data.body.length,
+      pages =
+        this.resultsPerPage === this.data.body.length
+          ? 1
+          : Math.floor(results / this.resultsPerPage) + 1;
+
+    return (
+      <Pagination
+        ref="pagination"
+        compact={
+          this.data.config.style.portal || this.data.config.pagination.compact
+        }
+        currentPage={this.currentPage}
+        pages={pages}
+        results={results}
+        pageSize={!this.data.config.style.portal}
+        pageJumper={!this.data.config.style.portal}
+      />
+    );
+  }
+
+  checkSelectAllCheckbox() {
+    const selectAllCheckbox = document.querySelector(
+      `.${DATA_TABLE_CLASSES.HEAD} .${DATA_TABLE_CLASSES.CELL} input`
+    ) as HTMLInputElement;
+
+    if (selectAllCheckbox) {
+      const isSelected = (row: DataTableRow) =>
+        this.selectedRows.includes(`row-${row.id}`);
+
+      selectAllCheckbox.checked = !!this.slicedData.every(isSelected);
+    }
+  }
+
+  sliceData(data: Record<string, any>) {
+    if (data.length > this.resultsPerPage) {
+      const arrayStartIndex = (this.currentPage - 1) * this.resultsPerPage,
+        arrayEndIndex =
+          (this.currentPage - 1) * this.resultsPerPage + this.resultsPerPage;
+
+      return data.slice(arrayStartIndex, arrayEndIndex);
+    }
+  }
+
+  detectScreenBreakpoint() {
+    const currentScreenSize = window.innerWidth;
+    let screenBreakpoint: DataTableScreenBreakpoints = 'xl';
+
+    switch (true) {
+      case currentScreenSize > 1200:
```

Comment:
@janibolkvadze Breakpoint sizes should be constants and global for all Chi project.